src/services/llm/chat_storage_service.ts

```diff
@@ -0,0 +1,217 @@
+import notes from '../notes.js';
+import sql from '../sql.js';
+import attributes from '../attributes.js';
+import type { Message } from './ai_interface.js';
+
+interface StoredChat {
+    id: string;
+    title: string;
+    messages: Message[];
+    noteId?: string;
+    createdAt: Date;
+    updatedAt: Date;
+}
+
+/**
+ * Service for storing and retrieving chat histories
+ * Chats are stored as a special type of note
+ */
+export class ChatStorageService {
+    private static readonly CHAT_LABEL = 'triliumChat';
+    private static readonly CHAT_ROOT_LABEL = 'triliumChatRoot';
+    private static readonly CHAT_TYPE = 'code';
+    private static readonly CHAT_MIME = 'application/json';
+
+    /**
+     * Get or create the root note for all chats
+     */
+    async getOrCreateChatRoot(): Promise<string> {
+        const existingRoot = await sql.getRow<{noteId: string}>(
+            `SELECT noteId FROM attributes WHERE name = ? AND value = ?`,
+            ['label', ChatStorageService.CHAT_ROOT_LABEL]
+        );
+
+        if (existingRoot) {
+            return existingRoot.noteId;
+        }
+
+        // Create root note for chats
+        const { note } = notes.createNewNote({
+            parentNoteId: 'root',
+            title: 'AI Chats',
+            type: 'text',
+            content: 'This note contains your saved AI chat conversations.'
+        });
+
+        attributes.createLabel(
+            note.noteId,
+            ChatStorageService.CHAT_ROOT_LABEL,
+            ''
+        );
+
+        return note.noteId;
+    }
+
+    /**
+     * Create a new chat
+     */
+    async createChat(title: string, messages: Message[] = []): Promise<StoredChat> {
+        const rootNoteId = await this.getOrCreateChatRoot();
+        const now = new Date();
+
+        const { note } = notes.createNewNote({
+            parentNoteId: rootNoteId,
+            title: title || 'New Chat ' + now.toLocaleString(),
```

Comment:
does it make sense to have the "default fallback string" translatable?

Reply:
Made these strings translateable